src/hooks/useTitleChanges.js

```diff
@@ -0,0 +1,42 @@
+import { useMemo, useEffect, useCallback } from 'react'
+import { getAppFullName } from 'lib/utils.js'
+
+function useTitleChanges({ noteId, title, setTitle, serviceClient }) {
+  // Title change because of a local change from the editor
+  const onLocalTitleChange = useCallback(
+    serviceClient
+      ? e => {
+          const modifiedTitle = e.target.value
+          if (title != modifiedTitle) {
+            setTitle(modifiedTitle)
+            serviceClient.setTitle(noteId, modifiedTitle)
+          }
+        }
+      : () => {},
+    [noteId, title, setTitle, serviceClient]
+  )
+  // Title change because of a remote change from the stack
+  useEffect(
+    () => {
+      serviceClient &&
+        serviceClient.onTitleUpdated(noteId, modifiedTitle => {
+          if (title != modifiedTitle) {
+            setTitle(modifiedTitle)
+          }
+        })
+    },
+    [title, setTitle, serviceClient]
+  )
```

Comment:
There are two different strategies here to deal with a potentially false serviceClient, inside the function or outside, could this be refactored so that only 1 techniques is used ?

Reply:
Could the code be refactored so that the hooks do not have to deal with the presence of serviceClient ? It seems cumbersome and error prone.

Reply:
I've tried a few things, but with no success. I tried to first explode the logic to simplify Editor. We should be able to change the content of each hook in a future step